src/speculators/data_generation/preprocessing.py

```
@@ -500,56 +305,165 @@ def _parse_conv_tools(conv_tools: object, idx: int) -> list | None:
+    parsed_tools = _parse_conv_tools(conv_tools, idx)
+    try:
+        return _render_boundary_rows(
+            normalized_conv,
+            render_endpoint,
+            max_length,
+            tools=parsed_tools,
+        )
+    # One row the render endpoint or boundary derivation can't handle must
+    # not kill the run. The failure modes can't be enumerated -- templates
+    # are swappable and raise arbitrary types -- so catch broadly and skip.
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)